File: Pokecard/app/src/main/java/com/example/lpiem/pokecard/Accueil.java
package com.example.lpiem.pokecard;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;

import android.support.v7.app.AppCompatActivity;

import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;

public class Accueil extends AppCompatActivity {
    Button button;
    FirebaseAuth mAuth;
    FirebaseAuth.AuthStateListener mAuthStateListener;

    @Override
    protected void onStart() {
        super.onStart();
        mAuth.addAuthStateListener(mAuthStateListener);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_accueil);

        button=findViewById(R.id.btn_sedeconnecter);
        mAuth= FirebaseAuth.getInstance();


        mAuthStateListener=new FirebaseAuth.AuthStateListener() {
            @Override
            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                if (firebaseAuth.getCurrentUser()==null) {
                    startActivity(new Intent(Accueil.this, Connexion.class));
                }   }
        };
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                    mAuth.signOut();
            }
        });

    }



}

File: Pokecard/app/src/main/java/com/example/lpiem/pokecard/Accueil.kt
/*package com.example.lpiem.pokecard

import android.content.Intent
import android.os.Bundle



import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

//import com.google.firebase.auth.FirebaseAuth

class Accueil : AppCompatActivity() {
    lateinit var button: Button
    //lateinit var mAuth: FirebaseAuth
    lateinit var mAuthStateListener: FirebaseAuth.AuthStateListener

    override fun onStart() {
        super.onStart()
        mAuth.addAuthStateListener(mAuthStateListener)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_accueil)
        button=findViewById(R.id.btn_sedeconnecter)

        mAuth = FirebaseAuth.getInstance()


        mAuthStateListener = FirebaseAuth.AuthStateListener { firebaseAuth ->
            if (firebaseAuth.currentUser == null) {
                val intent= Intent(this, Connexion::class.java)
                startActivity(intent)
            }
        }
        button.setOnClickListener { mAuth.signOut() }

    }


}
*/